Nodejs/Tests/TestAdapter.Tests/DefaultTestDiscovererTests.cs

```diff
@@ -0,0 +1,130 @@
+            }
+
+            var expectedCopy = new List<TestCase>(expected);
+            foreach (var testCase in actual)
+            {
+                var found = expectedCopy.Find(x =>
+                    x.FullyQualifiedName == testCase.FullyQualifiedName
+                    && x.DisplayName == testCase.DisplayName
+                    && x.LineNumber == testCase.LineNumber
+                    && x.ExecutorUri == testCase.ExecutorUri
+                    && this.PathEquals(x.Source, testCase.Source)
+                    && this.PathEquals(x.CodeFilePath, testCase.CodeFilePath)
+                    && string.Equals(x.GetPropertyValue<string>(JavaScriptTestCaseProperties.TestFramework, default), testCase.GetPropertyValue<string>(JavaScriptTestCaseProperties.TestFramework, default), StringComparison.OrdinalIgnoreCase) // For some reason, mocha test framework is all lowercase.
+                    && this.PathEquals(x.GetPropertyValue<string>(JavaScriptTestCaseProperties.WorkingDir, default), testCase.GetPropertyValue<string>(JavaScriptTestCaseProperties.WorkingDir, default))
+                    && this.PathEquals(x.GetPropertyValue<string>(JavaScriptTestCaseProperties.ProjectRootDir, default), testCase.GetPropertyValue<string>(JavaScriptTestCaseProperties.ProjectRootDir, default))
+                    && this.PathEquals(x.GetPropertyValue<string>(JavaScriptTestCaseProperties.TestFile, default), testCase.GetPropertyValue<string>(JavaScriptTestCaseProperties.TestFile, default))
+                    && this.PathEquals(x.GetPropertyValue<string>(JavaScriptTestCaseProperties.ConfigDirPath, default), testCase.GetPropertyValue<string>(JavaScriptTestCaseProperties.ConfigDirPath, default)));
+
+                if (found == null)
+                {
+                    Assert.Fail($"Expected does not have item: {JsonConvert.SerializeObject(testCase)}");
+                }
+
+                expectedCopy.Remove(found);
+            }
+        }
+
+        private void AssertProject(TestProjectFactory.ProjectName projectName, int expectedTests)
+        {
+            // Arrange
+            var actual = new List<TestCase>();
+            var expected = TestProjectFactory.GetTestCases(projectName);
+
+            var discoverContext = new Mock<IDiscoveryContext>();
+            var messageLogger = new Mock<IMessageLogger>();
+            var testCaseDiscoverySink = new Mock<ITestCaseDiscoverySink>();
+
+            testCaseDiscoverySink.Setup(x => x.SendTestCase(It.IsAny<TestCase>()))
+                .Callback<TestCase>(x => actual.Add(x));
+
+            var testSource = TestProjectFactory.GetProjectFilePath(projectName);
+            var sources = new List<string>() { testSource };
+
+            AssureNodeModules(TestProjectFactory.GetProjectDirPath(projectName));
+
+            var testDiscoverer = new DefaultTestDiscoverer();
```

Comment:
Out of curiosity, why was this type public?
I would expect that none of the types in the test adapter assembly are public, so IVT would be needed for the tests.

Reply:
I'm not sure I understand the question. You mean why `DefaultTestDiscoverer` is public? It implements `ITestDiscoverer` which is public as well, since I'm using it as well for testing, it make sense to make it public.

Reply:
Yeah, that was my question. I thought `ITestDiscoverer` wasn't our type, and even if it was, that doesn't mean the implementation needs to be public too.

My concern is that we have a public API here for no reason that someone could somehow take a dependency on.
We can use IVT to access internal types for testing purposes.

Reply:
Of course, I could be totally wrong about how our `ITestDiscoverer` implementations are invoked, in that they might have to be public for some reason I don't know about.

Reply:
You are correct, `ITestDiscoverer` is not our interface we just implement it and we don't explicitly require to be public nor the methods, afaik. The only reason we have it public is for the test project since we're referencing the class in it.

In regards to IVT, pardon my ignorance but what is that and how making it private/internal will allow us to access it?